drivers/st25r3911b/st25r3911b_interrupt.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <soc.h>
+#include <gpio.h>
+#include <logging/log.h>
+#include <drivers/st25r3911b_spi.h>
+#include <drivers/st25r3911b_interrupt.h>
+
+LOG_MODULE_DECLARE(st25r3911b);
+
+#define IRQ_REG_CNT 3
+
+static struct gpio_callback gpio_cb;
+static struct device *gpio_dev;
+
+static atomic_t irq_mask;
+static struct k_sem *sem;
+
+
+static void irq_pin_cb(struct device *gpiob, struct gpio_callback *cb,
+		       u32_t pins)
+{
+	k_sem_give(sem);
+}
+
+static int gpio_init(void)
+{
+	int err;
+
+	gpio_dev = device_get_binding(DT_GPIO_P0_DEV_NAME);
+	if (!gpio_dev) {
+		LOG_ERR("GPIO device binding error.");
+		return -ENXIO;
+	}
+
+	/* Configure IRQ pin */
+	err = gpio_pin_configure(gpio_dev, CONFIG_ST25R3911B_IRQ_PIN,
+				 GPIO_DIR_IN | GPIO_INT |
+				 GPIO_INT_EDGE | GPIO_INT_ACTIVE_HIGH);
+	if (err) {
+		return err;
+	}
+
+	gpio_init_callback(&gpio_cb, irq_pin_cb,
+			   BIT(CONFIG_ST25R3911B_IRQ_PIN));
+
+	err = gpio_add_callback(gpio_dev, &gpio_cb);
+	if (err) {
+		return err;
+	};
+
+	return gpio_pin_enable_callback(gpio_dev, CONFIG_ST25R3911B_IRQ_PIN);
+}
+
+u32_t st25r3911b_irq_read(void)
+{
+	u8_t data[IRQ_REG_CNT] = {0};
+	u32_t status = 0;
+	u32_t pin = 0;
+	int err;
+
+	/* In some cases, interrupts come so quickly that
+	 * the state of the pin will not change
```

Comment:
what is the expected delay? in range of us or ms?
maybe it would be good to add some safe fuse (e.g. max number of loop cycles)?

Reply:
I add a limitation to loop iteration, so now this should be safe. Delay is probably very low sometime this loop has about 2 iterations.